yt_dlp/extractor/francetv.py

```diff
@@ -193,28 +167,26 @@ def sign(manifest_url, manifest_id):
                         'format_id': format_id,
                     })
 
+            # XXX: what is video['captions']?
+            # XXX: video['spritesheets'] contains a list of thumbnail images (storyboard)
+
+        for f in formats:
+            if f.get('acodec') != 'none' and f.get('language') in ('qtz', 'qad'):
+                f['preference'] = -1
```

Comment:
I thought zero was the default. Seems a more natural choice. Wouldn’t it make sense to have named constants for that?

Also, would it not make sense to re-write the `language` key as well back to `fr`?

Reply:
> I thought zero was the default. Seems a more natural choice. Wouldn’t it make sense to have named constants for that?

If we were creating it from scratch, yes. But I dont think changing it is worth the headache

> Also, would it not make sense to re-write the language key as well back to fr?

I don't know id qtz/qad is standard notation. If not, I guess it makes sense

Reply:
`qtz` and `qad` belong in the private-use zone of ISO 639.

As an argument against, format selectors would no longer be able to tell regular and AD streams apart.

Reply:
AD should ideally be its own property. I've never seen `qad` on any other site so far, but I've seen Peacock use `en-AD`, which is actually wrong because it technically means English (Andorra). Most sites just have a separate property in the manifest or a suffix in the filename though.

Reply:
Indeed. DASH does it with an element `<Role schemeIdUri="urn:mpeg:dash:role:2011" value="description" />`.

By the way though, a named constant may be something as simple as 

```python
PREF_SECONDARY = -10
```
in some common file.